# Exploratory Data Analysis of Address table

In [2]:
import pandas as pd 

In [4]:
address = pd.read_csv("../data/output/address.csv")

address.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 603 entries, 0 to 602
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   address_id   603 non-null    int64  
 1   address      603 non-null    object 
 2   address2     0 non-null      float64
 3   district     600 non-null    object 
 4   city_id      603 non-null    int64  
 5   postal_code  599 non-null    float64
 6   phone        601 non-null    float64
 7   last_update  603 non-null    object 
dtypes: float64(3), int64(2), object(3)
memory usage: 37.8+ KB


### Missing data

Appears to be some missing data in the `district`, `postal_code` and `phone` columns. `address2` is missing completely    


In [9]:
# How many rows are we losing 

total_missing = (address["district"].isnull() | address["postal_code"].isnull() | address["phone"].isnull())
print(f"Total rows missing: {total_missing.sum()}")

Total rows missing: 7


> For a dataset of 603 rows, 7 missing rows (1.1%) is  certainly a small enough amount to be dropped without a second thought

### Duplicates


In [11]:
address.duplicated().sum()

np.int64(0)

It is apparent there are no duplicated records in the address table